doc/side-by-side-CTL-and-Plutus-comparison.md

```diff
@@ -0,0 +1,323 @@
+# Side by side comparison of CTL and Plutus off-chain contracts
+
+We are going to compare two different off-chain contracts 
+both in Plutus and CTL. 
+
+For this discussion, we need to go through an overview of 
+both of them.
+
+
+## About `Contract` in CTL and Plutus
+
+The purpose of `Contract` in CTL is to bring the capabilities of 
+`QueryMExtended` to the public API. 
+
+The current definition of `Contract` in CTL is :
+
+```PureScript
+type QueryConfig (r :: Row Type) =
+  { some_internal_parameters 
+  | r
+  }
+
+type DefaultQueryConfig = QueryConfig ()
+
+type QueryM (a :: Type) = ReaderT DefaultQueryConfig (LoggerT Aff) a
+
+type QueryMExtended (r :: Row Type) (a :: Type) =
+  ReaderT 
+    (QueryConfig r)
+    (LoggerT Aff)
+    a
+
+newtype Contract (r :: Row Type) (a :: Type) = Contract (QueryMExtended r a)
+```
+
+
+In CTL we have a general environment type `forall r . QueryConfig r`, it stores
+the needed parameters to connect to an `Ogmios` server  and
+uses the PureScript native row polymorphism to make it extensible for both 
+CTL developers and users.
+
+The parameter `a` as in `Plutus` refers to a return value wrapped by `Contract`.
+
+
+Note that in Plutus right now we have the following definition for `Contract`:
+
+```Haskell
+type ContractEffs w e =
+    '[ Error e
+    ,  LogMsg Value
+    ,  Writer w
+    ,  Checkpoint
+    ,  Resumable PABResp PABReq
+    ]
+
+type ContractEnv = (IterationID, RequestID)
+
+newtype Contract w (s :: Row *) e a = Contract { unContract :: Eff (ContractEffs w e) a }
+  deriving newtype (Functor, Applicative, Monad)
+```
+
+The Plutus `Contract` environment is specialized to just two values and is fixed. 
+Also, Plutus `Contract` uses a phantom type `s` to contract schema 
+and parameters `w` for a writer and `e` for errors. 
+In the case of CTL we don't have the contract schema parameter or the writer 
+parameter since CTL definition allows performing arbitrary effects. 
+This is possible  since the definition of `LoggerT` is: 
+
+```PureScript
+newtype LoggerT m a = LoggerT ((Message -> m Unit) -> m a)
+```
+
+The use of `Aff` inside `LoggerT` allows us to use asynchronous effects 
```

Comment:
We could easily remove `LoggerT` and the stack would still hold (up to logging changes). You can use `Effect.Class.Console (log)` to log to console.

There is no need for writer as we can perform arbitrary async. effects since we have `Aff` at the bottom of the stack?